packages/medusa/src/services/product.js

```diff
@@ -182,23 +181,21 @@ class ProductService extends BaseService {
 
     const { q, query, relations } = this.prepareListQuery_(selector, config)
 
+    let products
+    let count
     if (q) {
-      const qb = this.getFreeTextQueryBuilder_(productRepo, query, q)
-      const [raw, count] = await qb.getManyAndCount()
-
-      const products = await productRepo.findWithRelations(
+      ;[products, count] = await productRepo.getFreeTextSearchResultsAndCount(
+        q,
+        query,
+        relations
+      )
+    } else {
+      ;[products, count] = await productRepo.findWithRelationsAndCount(
```

Comment:
suggestion: Remove `;` should not be necessary

Reply:
addressed in previous comment (recap: this is added by prettier as a protective measure)

Reply:
Yes, means that having the debate on the end of line semilicon is more in favour of having them ^^